src/helm/clients/vision_language/qwen_vlm_client.py

```diff
@@ -0,0 +1,157 @@
+                prompt_text += f"Picture {image_index}: <img>{media_object.location}</img>\n"
+                image_index += 1
+            elif media_object.is_type(TEXT_TYPE):
+                if media_object.text is None:
+                    raise ValueError("MediaObject of text type has missing text field value")
+
+                query.append({"text": media_object.text})
+                prompt_text += media_object.text
+            else:
+                raise ValueError(f"Unrecognized MediaObject type {media_object.type}")
+
+        completions: List[Sequence] = []
+        request_time: float = 0
+        request_datetime: Optional[int] = None
+        all_cached: bool = True
+
+        with htrack_block(f"Generating for prompt: {prompt_text}"):
+            for completion_index in range(request.num_completions):
+                try:
+
+                    def do_it() -> Dict[str, Any]:
+                        if request.model_deployment == "Qwen/Qwen-VL-Chat":
+                            response, _ = model.chat(tokenizer, query=tokenizer.from_list_format(query), history=None)
+                        else:
+                            inputs = tokenizer(tokenizer.from_list_format(query), return_tensors="pt")
+                            inputs = inputs.to(self._device)
+                            pred = model.generate(**inputs, **generation_args)
+                            response = tokenizer.decode(pred.cpu()[0], skip_special_tokens=False)
+                        return {"output": response}
+
+                    # Include the prompt and model name in the cache key
+                    cache_key = CachingClient.make_cache_key(
+                        raw_request={
+                            "completion_index": completion_index,
+                            "model": request.model,
+                            "prompt": generate_uid_for_multimodal_prompt(request.multimodal_prompt),
+                            **generation_args,
+                        },
+                        request=request,
+                    )
+                    result, cached = self.cache.get(cache_key, wrap_request_time(do_it))
+                except RuntimeError as model_error:
+                    return RequestResult(
+                        success=False, cached=False, error=str(model_error), completions=[], embedding=[]
+                    )
+
+                text: str = result["output"]
+                hlog(f"Generated text: {text}")
+
+                # Truncate the output text as the original Qwen includes the prompt in the output sequence
+                if request.model_deployment == "Qwen/Qwen-VL":
+                    text = text[len(prompt_text) :]
+                    text = text.replace(self.END_OF_TEXT_TOKEN, "")
+                    hlog(f"Truncated: {text}")
+
+                # Tokenize truncated text to get the list of tokens
+                completions.append(Sequence(text=text, logprob=0, tokens=[]))
```

Comment:
Why are we not setting the tokens here?

Reply:
We don't really use tokens, but updated with tokens.